lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -684,274 +967,238 @@ export class UserAgentApplication {
+   * - In Azure B2C, it is of the form https://&lt;instance&gt;/tfp/&lt;tenant&gt;/<policyName>/
+   * - Default value is: "https://login.microsoftonline.com/common"
+   * @param {User} user - The user for which the scopes are requested.The default user is the logged in user.
+   * @param {string} extraQueryParameters - Key-value pairs to pass to the STS during the  authentication flow.
+   * @returns {Promise.<string>} - A Promise that is fulfilled when this function has completed, or rejected if an error was raised. Resolved with token or rejected with error.
    */
-  private validateInputScope(scopes: Array<string>): string {
-    if (!scopes || scopes.length < 1) {
-      return "Scopes cannot be passed as an empty array";
-    }
-
-    if (!Array.isArray(scopes)) {
-      throw new Error("API does not accept non-array scopes");
-    }
+  // TODO: params to accept AuthRequest object instead
+  @resolveTokenOnlyIfOutOfIframe
+  acquireTokenSilent(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+      // Validate scopes
+      // TODO: Replace with new validation pattern
+      const isValidScope = this.validateInputScope(scopes);
+      if (isValidScope && !Utils.isEmpty(isValidScope)) {
+        // TODO: Reject with custom error here
+        // TODO: Is this always accessToken?
+        reject(ErrorCodes.inputScopesError + "|" + isValidScope);
+        return null;
+      } else {
+        // TODO: Remove this from the else block, it is unnecessary
+        if (scopes) {
+          scopes = this.filterScopes(scopes);
+        }
 
-    if (scopes.indexOf(this.clientId) > -1) {
-      if (scopes.length > 1) {
-        return "ClientId can only be provided as a single scope";
-      }
-    }
-    return "";
-  }
+        const scope = scopes.join(" ").toLowerCase();
+        const userObject = user ? user : this.getUser();
+        const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+        //if user is not currently logged in and no login_hint/sid is passed as an extraQueryParamater
+        if (!userObject && Utils.checkSSO(extraQueryParameters) && Utils.isEmpty(adalIdToken) ) {
+          this._logger.info("User login is required");
+          // TODO: Reject with custom error here
+          reject(ErrorCodes.userLoginError + Constants.resourceDelimeter + ErrorDescription.userLoginError);
+          return null;
+        }
+        //if user didn't passes the login_hint and adal's idtoken is present and no userobject, use the login_hint from adal's idToken
+        else if (!userObject && !Utils.isEmpty(adalIdToken)) {
+          const idTokenObject = Utils.extractIdToken(adalIdToken);
+          console.log("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
```

Comment:
why are we not using this._logger?

Reply:
Logging is un-uniform today. Error implementation is addressing most of them.